src/test/java/com/google/sps/RecommenderServletTest.java

```diff
@@ -176,118 +181,93 @@ public void setUp() {
-    // all 5 recommendations. The coordinate formed by (cost-rating, crowd-rating) should be closest
-    // to the coordinate of user preference for the first recommendation, then respectively for 
-    // the remaining 4 (lowest Euclidean distance)
-
+  
+  // Function to mock the request and response
+  public String helperFunction (String category, String costRating, String crowdRating) throws Exception {
     // Mock request and mock response
     HttpServletRequest request = mock(HttpServletRequest.class);
     HttpServletResponse response = mock(HttpServletResponse.class);
-    when(request.getParameter("category")).thenReturn("restaurants");
-    when(request.getParameter("cost-rating")).thenReturn("3");
-    when(request.getParameter("crowd-rating")).thenReturn("3");
+    when(request.getParameter("category")).thenReturn(category);
+    when(request.getParameter("cost-rating")).thenReturn(costRating);
+    when(request.getParameter("crowd-rating")).thenReturn(crowdRating);
     StringWriter stringWriter = new StringWriter();
     PrintWriter writer = new PrintWriter(stringWriter);
     when(response.getWriter()).thenReturn(writer);
     myServlet.doGet(request, response);
     writer.flush();
+    return(stringWriter.toString());
+  }
+
+  @Test
+  public void allFiveReturned() throws Exception {
+    // When there are exactly five recommendations of that category, the algorithm should return  
+    // all 5 recommendations. The coordinate formed by (cost-rating, crowd-rating) should be closest
+    // to the coordinate of user preference for the first recommendation, then respectively for 
+    // the remaining 4 (lowest Euclidean distance)
     // 'Top 5' recommendations, ordered using the Euclidean distance (ED) algorithm
     Collection<Recommendation> expected = Arrays.asList(
       new Recommendation("Restaurant C", "restaurants", 53.0, -0.3, "A wonderful place to eat", 3, 3), // ED = 0
-      new Recommendation("Restaurant B", "restaurants", 52.0, -0.2, "A lovely place to eat", 2, 2), // ED = 1
-      new Recommendation("Restaurant D", "restaurants", 54.0, -0.4, "A charming place to eat", 4, 4), // ED = 1
-      new Recommendation("Restaurant A", "restaurants", 51.0, -0.1, "A great place to eat", 1, 1), // ED = sqrt(8) = 2.828...
-      new Recommendation("Restaurant E", "restaurants", 55.0, -0.5, "A popular place to eat", 5, 5) // ED = sqrt(8) = 2.828...
+      new Recommendation("Restaurant D", "restaurants", 54.0, -0.4, "A charming place to eat", 4, 3), // ED = 1
+      new Recommendation("Restaurant B", "restaurants", 52.0, -0.2, "A lovely place to eat", 2, 2), // ED = 2
+      new Recommendation("Restaurant A", "restaurants", 51.0, -0.1, "A great place to eat", 1, 2), // ED = 5
+      new Recommendation("Restaurant E", "restaurants", 55.0, -0.5, "A popular place to eat", 5, 5) // ED = 8
     );
+    String stringWriter = helperFunction("restaurants", "3", "3");
```

Comment:
Same here, please update the variable name to something like `personalisedRecommendations` or `result`

Reply:
Will do!